tests/test_project.py

```diff
@@ -2458,7 +2463,14 @@ def test_project_schema_versions(self):
     def test_project_schema_version_migration(self, implicit_version):
         from signac.contrib.migration import apply_migrations
 
-        config = read_config_file(self.project.fn("signac.rc"))
+        # First need to move the v2 config file to a suitable v0/v1 file.
```

Comment:
Schema version 0 doesn't exist except for our testing purposes, right? I think we only have schema version v1 in the wild.

Reply:
I think I'd rather hard-code a config here and use a specific filename in a temporary directory, without reference to the existing v2 schema or any other project initialization logic. That is, mock the schema v1 project construction with a few lines of Python so that we don't have to engineer a "roll-back" approach for testing future schema versions and we have something constant that lives outside our signac implementation. Roughly like:
```python
with TemporaryDirectory():
    with open("signac.rc", "w") as f:
        f.write(textwrap.dedent("""\
            project = asdf
            workspace_dir = workspace
            schema_version = 1"""
        )
```

Reply:
> I think we only have schema version v1 in the wild.
Yeah there's no real version 0. The default version assigned when a project doesn't have a version is 1.

> I think I'd rather hard-code a config here and use a specific filename in a temporary directory
Good call, I agree that would be better.